depthopt:
  rate: 1
  artificial_pointcloud: false

  topics:
    queue_size: 1000
    pointcloud_sub: /camera/depth/pointcloud
    ground_truth_sub: /vicon/depth_camera/depth_camera
    output_prefix: /depthopt/outputs
    # options: [history, error_history, final_error, relative_ground_truth, pointcloud_old, pointcloud_predicted, pointcloud_next, misc]
    outputs:
      - estimate
      - relative_ground_truth
      - history
      - pointcloud_old
      - pointcloud_predicted
      - pointcloud_next

  ground_truth_sync:
    queue_length: 1000
    ms_offset: 0

  window:
    mode: downsample
    size: 15

  outputs:
    sample_time_normalize: false
    optical_to_camera_frame: [-0.500, 0.500, -0.500, 0.500]

  optimizer:
    step_sizes: [0.0005, 0.0005]
    norm_weights: [1, 1]
    threshold: 0.0
    max_iterations: 50
    initialization_type: last   # options: {identity, last, external [not implemented]}
    record_history: true        # time consuming, use only for debug
    disable: false              # do not perform optimization, only collect data and test
    
    local_min_heuristics:       # after first optimization, draw gaussian samples around the result and retry optimization
      count: 0                  # number of new reinitializations
      scatter: 0.05             # variance of normal distribution to draw from
      enable: false
    
    cost:
      batch_size: 50          # number of points where landscape is evaluated
      stochastic: true
      reshuffle_batch_indexes: false

    landscape:
      measure_radius: 0.7
      smooth_radius: 0.001
      precision: 1                # number of particles in the montecarlo estimate
      decimation: 1              # pointcloud elements decimation factor
      clip_area:                  # remove points from pcl farther than max and lower than min 
        min: 0
        max: 1.5
      batch_size: 500            # number of points randomly draw from pcl used for landscape generation
      stochastic: true
  mod_flow:
    debug:
      enable: false
      #only_channels: [windowed_pointcloud]
      exclude_channels: [ground_truth_source]